app/controllers/users_controller.rb

```diff
@@ -1,11 +1,12 @@
 class UsersController < ApplicationController
 
-  def show
-    @user = User.find_by_id params[:id]
-    unless @user
-      flash[:warning] = t "not_found_user"
-      redirect_to root_path
-    end
+  before_action :find_user, only: [:show, :edit, :update]
+  before_action :logged_in_user, except: [:new, :create, :show]
```

Comment:
cái before_action chạy từ trên xuống nên sắp xếp cho đúng thứ tự đi, phải check log in hay chưa đầu tiên chứ